src/sentry/search/base.py

```diff
@@ -14,36 +14,13 @@
 
 
 class SearchBackend(Service):
-    __all__ = frozenset([
-        'query',
-    ])
+    __read_methods__ = frozenset(['query'])
+    __write_methods__ = frozenset()
+    __all__ = __read_methods__ | __write_methods__
 
     def __init__(self, **options):
         pass
 
-    def query(
-        self,
-        project,
-        query=None,
-        status=None,
-        tags=None,
-        bookmarked_by=None,
-        assigned_to=None,
-        first_release=None,
-        sort_by='date',
-        age_from=None,
-        age_to=None,
-        unassigned=None,
-        date_from=None,
-        date_to=None,
-        cursor=None,
-        limit=100
-    ):
-        """
-        The return value should be a CursorResult.
-
-        The limit here is a soft input limit, which gets trimmed by the Cursor.
-        This means the backend should query limit + 2 and return that within the
-        CursorResult.
```

Comment:
This comment seemed to be at the wrong level of abstraction (it's not relevant to the `SequencePaginator`, for instance) so I removed it.